docker/gcp_pudl_etl.sh

```diff
@@ -281,10 +283,16 @@ if [[ $ETL_SUCCESS == 0 ]]; then
         zenodo_data_release "$ZENODO_TARGET_ENV" 2>&1 | tee -a "$LOGFILE"
         ZENODO_SUCCESS=${PIPESTATUS[0]}
     fi
+    # If running a tagged release, ensure that outputs can't be accidentally deleted
+    # It's not clear that an object lock can be applied in S3 with the AWS CLI
+    if [[ "$GITHUB_ACTION_TRIGGER" == "push" && "$BUILD_REF" == v20* ]]; then
+        gcloud storage objects update "gs://pudl.catalyst.coop/$BUILD_REF/*" --temporary-hold 2>&1 | tee -a "$LOGFILE"
+        GCLOUD_TEMPORARY_HOLD_SUCCESS=${PIPESTATUS[0]}
+    fi
```

Comment:
This should prevent the accidental deletion of versioned outputs after they've been distributed, on GCS at least.